pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -126,7 +127,7 @@ in (buildEnv {
       rm "$link"
       makeWrapper "$target" "$link" \
         --inherit-argv0 \
-        --prefix PATH : "${lib.makeBinPath [ coreutils gawk ]}:$out/bin" \
+        --prefix PATH : "${lib.makeBinPath (lib.optional needsGhostscript ghostscript)}:$out/bin" \
```

Comment:
Is removing `coreutils` and `gawk` safe?

Reply:
It's a gamble: I am betting that `tests.texlive.binaries` is able to flag all the important dependencies (now it even checks for `command not found`).

Reply:
Apparently there used to be an issue: https://github.com/NixOS/nixpkgs/commit/01e4d1a67f6062a6c59046dbc4461659ccff8031
Unfortunately we didn't get a regression test contributed at that time.

Reply:
Ah, this is also a rebase problem, coreutils and gawk should have disappeared in the main bin containers commit. I should rebase again.

Reply:
Spotted a bug: the `:` between ghostscript and `$out/bin` must be part of the `lib.optional`, otherwise you can get an empty entry in `PATH`. Fixed in new update.

Reply:
The uses of awk and coreutils are sometimes only in codepaths not executed in the tests -- grepping the tree for g?awk, sed and grep (with a lot of files excluded, there might be more), I find a couple of scripts where they're missing:
awk: 
 * dvipsconfig (addpsctrl)
 * listbib
 * listings-ext
 * pdfjam
 * texlive-scripts-extra (texlinks,texconfig)
 * texlive-scripts (mktexpk, mktexlsr)
 
grep:
 * git-latexdiff
 * l3build
 * logicpuzzle
 * ltxfileinfo
 * pdfjam
 * pdftex-quiet
 * pdfxup
 * texlive-scripts-extra (texlinks,texconfig)
 * texlive-scripts (mktextfm,mktexpk)
 
 sed:
 * bibexport
 * checklistings
 * dtxgen
 * git-latexdiff
 * installfont
 * latexdiff (latexdiff-vc.pl)
 * logicpuzzle
 * pagelayout (textestvis)
 * pdfjam
 * pgfplots (pgf2pdf.sh)
 * pst2pdf
 * simpdftex

I haven't checked for coreutils. Ideally, we still have all the dependencies inside the bin containers to allow standalone usage without a combined texlive env -- but also blanket-adding coreutils, grep , sed and awk in the wrappers to be on the safe side doesn't really hurt. Reallistically no one will have a system without them anyway, so the increase in closure size for combined environments where nothing pulls them in (which don't really make sense anyway) doesn't really matter.

Reply:
Let me say, I agree with this, it's fairly reasonable to add coreutils, gawk, sed by default. I'll rebase accordingly at the end of this week. I would ask you to do it to speed things up (I am travelling right now) but you probably want me around after merging anyway, in case something goes horribly wrong.

Reply:
Updated with coreutils, awk, grep, sed, please check that it looks ok!